src/protogui/WriteFile.java

```diff
@@ -257,33 +260,102 @@ public int generateColumn2Numbers(int problemNumbers)
         return numberColumn2;
     }
     
-    public String generateCarriedOnes(int num1, int num2)
+    public String generateCarriedNumbers(int num1, int num2)
     {
-        boolean carriedOne = false;
-        String solutionOnes = "&nbsp;&nbsp;";
+        String carriedDigits = "&nbsp;&nbsp;";
         while(num1 > 0)
+        {
+            int dig1 = num1 % 10;
+            int dig2 = num2 % 10;            
+            
+            
+            num1 = num1 / 10;
+            num2 = num2 / 10;
+            int sub = dig1 - dig2;
+            
+            if(sub < 0)
             {
-                int digit1 = num1 % 10;
-                int digit2 = num2 % 10;
-                
+                dig1 = num1 % 10;
                 num1 = num1 / 10;
                 num2 = num2 / 10;
                 
-                int solution = digit1 + digit2;
-                if(carriedOne)//adding 1 if the last digits were greater than 10
-                    solution++;
+                if(dig1 == 0)
+                    carriedDigits = 9 + carriedDigits;
+                else
+                    carriedDigits = (dig1 - 1) + carriedDigits;
                 
-                if(solution > 9)
+                
+                while(dig1 == 0)//looking for number to burrow a 1 from
                 {
-                    solutionOnes = "1" + solutionOnes;
-                    carriedOne = true;                    
+                    
+                    dig1 = num1 % 10;
+                    num1 = num1 / 10;
+                    num2 = num2 / 10;
+                    if(dig1 == 0)
+                    {
+                        carriedDigits = 9 + carriedDigits;
+                        
+                    }
+                    else
+                    {
+                        carriedDigits = (dig1 - 1) + carriedDigits;
+                    }
+                    
+                    
                 }
-                else
-                {
-                    solutionOnes = "&nbsp;&nbsp;" + solutionOnes;
-                    carriedOne = false;
-                } 
             }
+            else
+            {
+                carriedDigits = "&nbsp;&nbsp;" + carriedDigits;
+            }
+        }
+        return carriedDigits;
+    }
+    
+    public void generateCarriedSubtraction(int problemNum, int num1 , int num2, int numberLength, int result) throws IOException
+    {                         
+        String padding = generatePadding();
+        
+        String carriedDigits = generateCarriedNumbers(num1,num2);
```

Comment:
Very Slick embedded style tags on these solutions. I would rather just implement this than worrying about formating on an external style sheet if you are more comfortable doing this it will make the process alot easier.